src/APIActionCreator.es6.js

```diff
@@ -117,6 +117,11 @@ export default class APIActionCreator extends ActionCreator {
 			this
 		);
 
+		invariant(
+			withCredentials === undefined || typeof withCredentials === 'boolean',
```

Comment:
```js
const {route, method, withCredentials=false} = option
```
Could be done instead of the conditional in the invariant; I don't have a preference. Neither would require every method to specify `withCredentials`.